Mathlib/LinearAlgebra/Finsupp/Defs.lean

```diff
@@ -167,6 +167,10 @@ def lcomapDomain (f : α → β) (hf : Function.Injective f) : (β →₀ M) →
   map_add' x y := by ext; simp
   map_smul' c x := by ext; simp
 
+theorem lcomapDomain_leftInverse (f : α → β) (hf : Function.Injective f) :
+    Function.LeftInverse (lcomapDomain (R := R) (M := M) f hf) (mapDomain f) :=
+  fun _ ↦ Finsupp.ext fun a ↦ by simp [mapDomain_apply hf]
```

Comment:
Is `Finsupp.mapDomain_comapDomain` this lemma, or is that the other direction?

Reply:
It's the other direction (which only holds with the assumption on the range).

Reply:
Can you add `Finsupp.comapDomain_mapDomain` as a lemma then?

Reply:
Done.